### PR TITLE
adds management for drainer config timeout condition

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/3708. Nothing should change in actual behaviour for KVM because we already had the fallback implemented here. We just moved it to `node-operator` so all providers can leverage the same behaviour based on their CR status conditions.

### CODE REVIEW

**Comment by xh3b4sd** [comment]
Successfully tested on `gorgoth`.

File: service/controller/v14/resource/pod/delete.go [hunk 1]
```diff
@@ -21,24 +20,25 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "debug", "looking for the current version of the reconciled pod in the Kubernetes API")
```
Comment (xh3b4sd) [inline service/controller/v14/resource/pod/delete.go]: I moved the logging to their blocks. I also fixed the structure to have the level and message keys.

File: service/controller/v14/resource/pod/delete.go [hunk 2]
```diff
@@ -55,8 +55,8 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		}
 	}
 
-	if !forcePodCleanup(currentPod) {
```
Comment (xh3b4sd) [inline service/controller/v14/resource/pod/delete.go]: We use the fallback defined by the timeout status condition now.

**Review by MarcelMue — APPROVED** [review]
LGTM, the diff is kind of fucky for some reason.

File: service/controller/v14/resource/pod/delete.go [hunk 3]
```diff
@@ -83,20 +84,77 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
...
+						Endpoint: apiEndpoint,
+					},
+					ID: pod.GetNamespace(),
+				},
+				Node: corev1alpha1.DrainerConfigSpecGuestNode{
+					Name: pod.GetName(),
+				},
+			},
+			VersionBundle: corev1alpha1.DrainerConfigSpecVersionBundle{
+				Version: "0.1.0",
```
Comment (MarcelMue) [inline service/controller/v14/resource/pod/delete.go]: Something I just thought about:
How do we plan to manage this? What happens if the version would move up, how do we remember to change it here?

Comment (xh3b4sd) [reply service/controller/v14/resource/pod/delete.go]: I am not sure. This is definitely something we should address. How is the version dispatching managed with `chart-operator`? Don't we have the same issue there that some version bundle versions have to be forwarded?

Comment (MarcelMue) [reply service/controller/v14/resource/pod/delete.go]: Yes we do, but we don't address it either.
I tried to get at this in slack: https://gigantic.slack.com/archives/C2MS2VB37/p1531472745000275
This is quite concerning to me, it is bound to the `Versioning of Operators` story in my opinion but it wasn't on my radar at all.

Comment (xh3b4sd) [reply service/controller/v14/resource/pod/delete.go]: Lets make sure we track this somewhere. Addressing it with the release operator makes sense. If there is need lets get together and have a brainstorm.

**Pull request merged**